### ISSUE TITLE
Wrong title in context menu

### ISSUE THREAD

**Issue by sss-user** [CanisLupus/swift-selection-search#217]
If I r-click a URL, the context menu shows the text contents of the clipboard rather than the URL.

For example, let's say I have previously visited a Mozilla web page, and now I wish to send the "choose a different type" URL to Virustotal. I do this by r-clicking the new URL and selecting Virustotal from the drop-down list. 

However, instead of ...

search: "https://github.com/CanisLupus/swift-selection-search/issues/new/choose...

... I see ...

search: "https://addons.mozilla.org/en-US/firefox/addon/context-search-we/...

If I continue, then the desired URL is correctly submitted to Virustotal rather than the clipboard contents. So nothing is broken, it's just that the menu is confusing.

If I have previously searched for some text, then this text appears in the menu rather than the desired URL, but the search still works properly.

**Comment by CanisLupus** [comment]
Hi there! I'm sorry but I couldn't understand what you mean. Let's see...

1. As of now, SSS does not support right clicking a URL to search. It's easy to implement and was requested, but sadly the implementation has some inconsistencies that are pretty much impossible to "solve", so I left it like this (hard to explain :)).

2. The text that appears in the context menu should always be the selected text (i.e. not the clipboard and not the URL text). If the SSS menu appears in your context menu, it's likely that you have some text selected in the page, unless I'm missing something.

3. At least the first link you posted seems unrelated to your description. Might have been a mistake.

In summary, do you mean that you see your clipboard's contents (not the selected text) in the SSS's context menu, and that the menu shows up when right clicking a link, even if no text is selected? Did I get that right? Could you provide some screenshots?


This is what you should see:

![image](https://user-images.githubusercontent.com/532708/98597894-964bee80-22d1-11eb-8b55-b36c26025c2c.png)

Regards,
Daniel

**Comment by sss-user** [comment]
Sorry for my confusing explanation.

Firstly, I should have said I'm using SPC rather than SSS. Does that make a difference?

Secondly, I don't select or highlight any text. I simply r-click a URL. 

For example, on this very page I r-click this URL in my opening post:

https://github.com/CanisLupus/swift-selection-search/issues/new/choose

The context menu then presents me with ...

search: "the_previous_URL_that_was_in_the_clipboard...

... or ...

search: "previous_unrelated_clipboard_text...

If I then select any of the search engines, the currently r-clicked URL is submitted to that search engine rather than the one in the clipboard or in the context menu.

I have no problem submitting the r-clicked URL to Virustotal.

My OS is Win 10 32-bit, and my browser is Edge (Chrome).



![SPC_wrong_URL_in_Context_Menu](https://user-images.githubusercontent.com/74175891/98630834-1aed4a00-2370-11eb-833c-c50a7a90b4ed.gif)
![SPC_wrong_text_instead_of_URL_in_Context_Menu](https://user-images.githubusercontent.com/74175891/98630854-2c365680-2370-11eb-9afa-ff232d4b2164.gif)
![SPC_search_engines](https://user-images.githubusercontent.com/74175891/98630874-35bfbe80-2370-11eb-929d-9ec6ed586d69.gif)
![SPC_search_engines_config](https://user-images.githubusercontent.com/74175891/98630905-46703480-2370-11eb-9731-849b4626298d.gif)

**Comment by CanisLupus** [comment]
No problem at all! So that was the source of my confusion. :) We can close this, of course.

Could you please point me to this "SPC"? The last screenshot is very similar to the design of SSS's own engines list and you mentioned a "fork", so I assume it's a fork of this (SSS) repo, modifying some things for Google Chrome (since SSS only exists for Firefox), but I'm having a hard time finding it anywhere. And it doesn't seem to be any of the forks that GitHub lists for SSS.

I'm closing this thread, but you'll still be able to reply without a problem.

Cheers!
Daniel

**Comment by sss-user** [comment]
This project acknowledges your own as the Original Extension:

YoshifumiFuyuno/Search-from-Popup-or-ContextMenu

https://github.com/YoshifumiFuyuno/Search-from-Popup-or-ContextMenu

Sorry once again.

**Comment by CanisLupus** [comment]
Ah, thanks so much for the link. I really couldn't find it. I see now that it's a new repository without being a real GitHub "fork", that's why I couldn't. :)

Yep, containing the reference to the original MIT license is really the only thing required to create a derivative of SSS, so it's all good! It also has a detailed list of the changes, which is interesting to see.

No problem, mistakes happen. ;)
